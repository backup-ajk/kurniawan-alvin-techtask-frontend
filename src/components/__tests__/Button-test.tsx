import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { Button } from '../';

it('should render Button successfully', () => {
  const mockFunction = jest.fn();

  const input = renderer
    .create(<Button label={'test'} onClick={mockFunction} />)
    .toJSON();
  expect(input).toMatchSnapshot();
});

it('should render Button style correctly', () => {
  const mockFunction = jest.fn();

  const renderedDefault = renderer.create(
    <Button label={'test'} onClick={mockFunction} />,
  );
  const buttonDefault = renderedDefault.root.findByType('button');
  expect(buttonDefault.children).toEqual(['test']);
  expect(buttonDefault.props.style.backgroundColor).toEqual('#666ee8');

  const renderedSuccess = renderer.create(
    <Button label={'success'} onClick={mockFunction} theme={'success'} />,
  );
  const buttonSuccess = renderedSuccess.root.findByType('button');
  expect(buttonSuccess.props.style.backgroundColor).toEqual('#28d094');

  const renderedDanger = renderer.create(
    <Button label={'danger'} onClick={mockFunction} theme={'danger'} />,
  );
  const buttonDanger = renderedDanger.root.findByType('button');
  expect(buttonDanger.props.style.backgroundColor).toEqual('#ff4961');
});

it('should handle hover state on button', () => {
  const mockFunction = jest.fn();
  const button = shallow(<Button label={'test'} onClick={mockFunction} />);

  expect(button.state('isHover')).toBe(false);
  button.simulate('mouseenter');
  expect(button.get(0).props.style).toHaveProperty(
    'backgroundColor',
    '#7c83eb',
  );

  expect(button.state('isHover')).toBe(true);
  button.simulate('mouseleave');
  expect(button.get(0).props.style).toHaveProperty(
    'backgroundColor',
    '#666ee8',
  );

  expect(button.state('isHover')).toBe(false);
});
