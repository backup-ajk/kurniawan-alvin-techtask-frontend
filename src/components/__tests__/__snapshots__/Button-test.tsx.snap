// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button successfully 1`] = `
<button
  onClick={[MockFunction]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundColor": "#666ee8",
      "borderColor": "#3a44e1",
      "borderRadius": "0.25rem",
      "color": "#fff",
      "cursor": "pointer",
      "fontSize": 16,
      "fontWeight": 600,
      "padding": "0.75rem 1rem",
      "textTransform": "uppercase",
    }
  }
>
  test
</button>
`;
